apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/RetryingMutateRowsTest.cs

```diff
@@ -183,6 +186,88 @@ public class RetryingMutateRowsTest
+
+            var request = new MutateRowsRequest
+            {
+                Entries =
+                {
+                    Mutations.CreateEntry("a", Mutations.DeleteFromRow()),
+                    Mutations.CreateEntry("b", Mutations.DeleteFromRow()),
+                    Mutations.CreateEntry("c", Mutations.DeleteFromRow())
+                }
+            };
+            var client = Utilities.CreateMutateRowsMockClient(
+                request,
+                entriesForInitialStream: new[]
+                {
+                    Utilities.CreateMutateRowsResponseEntry(0, Code.Ok),
+                    Utilities.CreateMutateRowsResponseEntry(1, Code.DeadlineExceeded),
+                    Utilities.CreateMutateRowsResponseEntry(2, Code.Ok)
+                },
+                entriesForRetryStreams: new[]
+                {
+                    // 1st retry response entries
+                    new[] { Utilities.CreateMutateRowsResponseEntry(0, Code.Ok) }
+                },
+                settings: settings);
+
+            bool exceptionOccurred = false;
+            try
+            {
+                await client.MutateRowsAsync(request);
+            }
+            catch (RpcException ex) when (ex.StatusCode == StatusCode.DeadlineExceeded)
+            {
+                exceptionOccurred = true;
+            }
+            Assert.True(exceptionOccurred);
+        }
+
+        [Fact]
+        public async Task RetryBeforeTotalExpiration()
+        {
+            var request = new MutateRowsRequest
+            {
+                Entries =
+                {
+                    Mutations.CreateEntry("a", Mutations.DeleteFromRow()),
+                    Mutations.CreateEntry("b", Mutations.DeleteFromRow()),
+                    Mutations.CreateEntry("c", Mutations.DeleteFromRow())
+                }
+            };
+            var client = Utilities.CreateMutateRowsMockClient(
+                request,
+                entriesForInitialStream: new[]
+                {
+                    Utilities.CreateMutateRowsResponseEntry(0, Code.Ok),
+                    Utilities.CreateMutateRowsResponseEntry(1, Code.DeadlineExceeded),
+                    Utilities.CreateMutateRowsResponseEntry(2, Code.Ok)
+                },
+                entriesForRetryStreams: new[]
+                {
+                    null, // A null entry will throw an Unavailable RpcException
+                    // 2nd retry response entries
+                    new[] { Utilities.CreateMutateRowsResponseEntry(0, Code.Ok) }
+                });
+
+            bool exceptionOccurred = false;
+            try
+            {
+                await client.MutateRowsAsync(request);
+            }
+            catch (RpcException ex) when (ex.StatusCode == StatusCode.DeadlineExceeded)
+            {
+                exceptionOccurred = true;
+            }
+            Assert.False(exceptionOccurred);
```

Comment:
Can we easily check that the clock is where we'd expect it to be?

Reply:
Or is there not actually a clock involved? I don't see one in the changes - is that already taken care of elsewhere?

Reply:
Yeah, I think we could do more in the future with mock clocks, and this is slightly hacky, but I didn't want to complicate the `Utilities.Create...MockClient` logic any more than necessary for now.